rust/src/bls/api.rs

```diff
@@ -256,61 +237,53 @@ pub unsafe extern "C" fn fil_private_key_generate() -> *mut types::fil_PrivateKe
 /// * `raw_seed` - a seed byte array with 32 bytes
 ///
 /// Returns `NULL` when passed a NULL pointer.
-#[no_mangle]
-pub unsafe extern "C" fn fil_private_key_generate_with_seed(
-    raw_seed: fil_32ByteArray,
-) -> *mut types::fil_PrivateKeyGenerateResponse {
-    let rng = &mut ChaChaRng::from_seed(raw_seed.inner);
+#[ffi_export]
+pub fn private_key_generate_with_seed(
+    raw_seed: &[u8; 32],
+) -> repr_c::Box<types::PrivateKeyGenerateResponse> {
+    let rng = &mut ChaChaRng::from_seed(*raw_seed);
 
     let mut raw_private_key: [u8; PRIVATE_KEY_BYTES] = [0; PRIVATE_KEY_BYTES];
     PrivateKey::generate(rng)
         .write_bytes(&mut raw_private_key.as_mut())
         .expect("preallocated");
 
-    let response = types::fil_PrivateKeyGenerateResponse {
-        private_key: fil_BLSPrivateKey {
+    let response = types::PrivateKeyGenerateResponse {
+        private_key: BLSPrivateKey {
             inner: raw_private_key,
         },
     };
 
-    Box::into_raw(Box::new(response))
+    repr_c::Box::new(response)
 }
 
 /// Sign a message with a private key and return the signature
 ///
 /// # Arguments
 ///
-/// * `raw_private_key_ptr` - pointer to a private key byte array
-/// * `message_ptr` - pointer to a message byte array
-/// * `message_len` - length of the byte array
+/// * `raw_private_key` - pointer to a private key byte array
+/// * `message` - pointer to a message byte array
```

Comment:
Those are not pointers anymore (at least you didn't use the word "pointer" in other cases).

Reply:
fixed